src/core/ext/transport/chttp2/transport/hpack_parser.cc

```diff
@@ -811,11 +811,11 @@ static grpc_error* parse_indexed_field_x(grpc_chttp2_hpack_parser* p,
    cached metadata corresponds with the index we are examining. */
 static grpc_mdelem get_precomputed_md_for_idx(grpc_chttp2_hpack_parser* p) {
   GPR_DEBUG_ASSERT(p->md_for_index.payload != 0);
-  GPR_DEBUG_ASSERT(static_cast<int64_t>(p->index) == p->precomputed_md_index);
   grpc_mdelem md = p->md_for_index;
   GPR_DEBUG_ASSERT(!GRPC_MDISNULL(md)); /* handled in string parsing */
   p->md_for_index.payload = 0; /* Invalidate cached md when index changes. */
 #ifndef NDEBUG
+  GPR_DEBUG_ASSERT(static_cast<int64_t>(p->index) == p->precomputed_md_index);
```

Comment:
This change seems unrelated to this PR?

Reply:
It moved down a few lines so I could keep it within a common `#ifndef DEBUG` that was already there rather than adding a new one. If you'd prefer, I can keep it in its original place and just put in a new `#ifndef` block

Reply:
But how is it related to this PR?

Reply:
Oh you're right; I'll revert this change. This got in because of a change I had attempted earlier which ended up not making sense.